utils/src/test/java/com/cloud/utils/net/MacAddressTest.java

```diff
@@ -51,10 +52,11 @@ public final void testMacAddressToLong() throws Exception {
     // TODO    public final void testToString() throws Exception {
     // TODO    public final void testGetMacAddress() throws Exception {
     // TODO    public final void testParse() throws Exception {
-    // TODO    public final void testMain() throws Exception {
-    // TODO    public final void testParseLong() throws Exception {
-    // TODO    public final void testParseInt() throws Exception {
-    // TODO    public final void testParseShort() throws Exception {
-    // TODO    public final void testParseByte() throws Exception {
 
+    public static void main(String[] args) {
```

Comment:
A "main" method in a test file? Why not create unit test cases instead?

Reply:
it was just a move, and the logic didn't seem to justify a unit test

Reply:
hmm.. So, what does justify a "main" method in the test class?

Reply:
:) a test mac addresses utility ???

Reply:
I still do not get it... I think we should not have "main" methods laying around. If it is a test worth having, it should be a unit test case. These "main" methods that we have, they all seem to be left overs that programmers "forget" when pushing code.

Reply:
I don't think I am making a strong argument but I do think main methods 'laying around' are very usefull for all sorts of things; validating ip (6v) addresses, validating mac addresses or encrypting/decrypting are but a few examples of command line tools that are usefull and one would like to have using the same codebase as the core system.

Reply:
We are having two discussions, I feel. I am talking about having cli utilities laying around. And yes, the location of this main is still not ideal. We can delete it in which case a user or dev can not manually check a mac address they concocted. this is not for verifying the method and also not very useful as cli util.

Reply:
Now I think I understood what you meant, but still I am skeptical with code that we might leave behind. This can be used as an excuse to maintain non-used code that we constantly see around. In my perspective a developer should be able to find tool to validate/execute small checks instead of building them.

Reply:
So what about a package with assorted cli utils? (i'll move from here)

Reply:
If we want some sort of CLI utils, it should be a component that one is able to build and use without being a developer. The way it is now, this class is only used by developers (if it is used at all) via their IDE.

Reply:
hm, I yield. I'd like to have a set of cli based on the exact same code that we use in the system, but this is definately not the place ;)

Reply:
I do like the idea of having some "extra" CLI tools to help administrators/operators (specially these two groups that sometimes get lost in ACS world). However, it needs to be well crafted and designed.